modules/xfem/doc/content/source/userobjects/InterfaceMeshCut2DUserObject.md

```diff
@@ -0,0 +1,17 @@
+# InterfaceMeshCut2DUserObject
+
+!syntax description /UserObjects/InterfaceMeshCut2DUserObject
+
+## Overview
+
+The `InterfaceMeshCut3DUserObject` is used to cut the 3D mesh with a 2D cutter mesh. The element type of the 2D cutter mesh must be `TRI3`. The nodes on the cutter mesh move at a velocity that is given by a function or `XFEMMovingInterfaceVelocity` UserObject.
```

Comment:
I am coming back to this markdown documentation after reviewing the actual source code... You definitely need to mention that the velocity is a lumped average over all nodal velocities.

Then a design question follows: should we at least add an option to turn velocity averaging on/off? I would imagine there will be cases where users want to prescribe a function that varies over space.

Reply:
@hugary1995 Only the nodes that are outside the domain will use the average velocity because they do not have any quantities to calculate the velocity.